src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -75,4 +77,18 @@ private void verifyBelowMaxNumTrips(String userId, Request request) {
             );
         }
     }
+
+    /**
+     * Checks that the given {@link MonitoredTrip} can be monitored
+     * (the underlying {@link org.opentripplanner.middleware.otp.response.Itinerary} has transit and no rentals).
+     */
+    private void checkItineraryCanBeMonitored(MonitoredTrip trip, Request request) {
+        if (!ItineraryUtils.itineraryCanBeMonitored(trip.itinerary)) {
+            logMessageAndHalt(
+                request,
+                HttpStatus.BAD_REQUEST_400,
+                "Only trips with an itinerary that includes transit and no rentals or ride hailing can be monitored."
+            );
+        }
```

Comment:
Instead of embedding the logic deep within ItineraryUtils, it might be better to have independent methods be invoked here so the error message can be tailored a little more easily and the logic is a bit more transparent. For example:

```
      if (!ItineraryUtils.itineraryHasTransit(itinerary)) {
            logMessageAndHalt(
                request,
                HttpStatus.BAD_REQUEST_400,
                "Monitored trip must reference an itinerary that includes transit legs."
            );
        }
       if (ItineraryUtils.itineraryHasRental(itinerary)) {
            logMessageAndHalt(
                request,
                HttpStatus.BAD_REQUEST_400,
                "Monitored trip must not reference an itinerary that includes legs using rental vehicles."
            );
        }
```

Reply:
Who should break the tie?

Reply:
If a trip cannot be monitored, I think there should be a single message saying all the conditions that should be met. That reminds me to update the swagger method description too...

Reply:
I like the idea of providing a list of criteria that must be met.

Reply:
@evansiroky please clarify what needs to change, thank you.

Reply:
I think what @evansiroky is suggesting is we collect a list of validation issues. So, it would be similar to the code I posted above, but the halt should only happen at the end of the method and it would include the full list of validation issues.

Reply:
Let's try this: https://github.com/ibi-group/otp-middleware/pull/91/commits/83405f10b950ed5b4fb6ca75ddfa1a1f34753d1e.